app/app/api/profiles/route.ts

```diff
@@ -0,0 +1,95 @@
+import { NextRequest, NextResponse } from "next/server";
+import prisma from "database/db";
+
+
+
+export async function GET(req: NextRequest) {
+  let profiles;
+
+  try {
+    profiles = await prisma.user.findMany({
+        select: {
+            id: true,
+            firstName: true,
+            lastName: true,
+            email: true,
+            userToUserRoles: {
+              select: {
+                role: {
+                    select: {
+                        name: true,
+                    }
+                },
+              },
+            },
+            image: true,
+            departmentMemberships: {
+              select: {
+                department: {
+                  select: {
+                    id: true,
+                    name: true,
+                  },
+                },
+                departmentPosition: true,
+                membershipEnd: true,
+                membershipStart: true,
+              },
+            },
+          },
+      });
+  } catch (error) {
+      return NextResponse.json({ error: error }, { status: 400 });
+  }
+
+  const new_profiles = prepareMembershipData(profiles);
+
+  return NextResponse.json({ profiles: new_profiles }, { status: 200 });
+}
+
+function prepareMembershipData(profiles: any) {
+  // not possible to order by nested fields. function required.
+  // orderBy: {
+  //     membership_end: 'desc',
+  //     membership_start: 'desc'
+  // }
+  // in Prisma2 https://github.com/graphql-nexus/nexus-plugin-prisma/issues/458
```

Comment:
This seems like a pretty old issue. At first glance, https://www.prisma.io/docs/orm/prisma-client/queries/filtering-and-sorting comes to mind. Incase this is too time intensive to do now, consider this comment resolved.